From 97bfd4f98c114158eff4bcfb91deb745789dce2a Mon Sep 17 00:00:00 2001
From: Neta Zur <neta@marvell.com>
Date: Thu, 3 Sep 2015 14:40:07 +0300
Subject: [PATCH 106/538] ap806: thermal: update divisor and fix registers
 addresses

- Thermal driver need to return temperature in millidegrees
- register were at bad offset

Change-Id: I23640709f4ea8dd4bb828a0a07ac59e77b3578e1
Signed-off-by: Neta Zur <neta@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23497
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 4 ++--
 drivers/thermal/armada_thermal.c              | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 32af34f..2c6ccc3 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -226,9 +226,9 @@
 				dma-coherent;
 			};
 
-			thermal@7f008c {
+			thermal@6f808c {
 				compatible = "marvell,armada-ap806-thermal";
-				reg = <0x7f008C 0x4>, <0x7f0084 0x4>;
+				reg = <0x6f808C 0x4>, <0x6f8084 0x4>;
 				status = "okay";
 			};
 		};
diff --git a/drivers/thermal/armada_thermal.c b/drivers/thermal/armada_thermal.c
index 44bea5c..4e3f2d4 100644
--- a/drivers/thermal/armada_thermal.c
+++ b/drivers/thermal/armada_thermal.c
@@ -302,7 +302,7 @@ static const struct armada_thermal_data armada_ap806_data = {
 	.temp_mask = 0x3ff,
 	.coef_b = 153400,
 	.coef_m = 425,
-	.coef_div = 1000,
+	.coef_div = 1,
 	.inverted = true,
 	.ops = &armada_ap806_ops,
 };
-- 
1.9.1

